### PR TITLE
Fix pd.NaT - Series

### LINKED ISSUE #19158: NaT - Series[timedelta64] returns NaT
Surfaced in #19139, not present in 0.22.0:

```
tdi = pd.TimedeltaIndex(['1 day', '2 day'], name='x')
ser = pd.Series(tdi)

>>> pd.NaT - ser
NaT
```

Expected:
```
>>> pd.NaT - ser
0   NaT
1   NaT
Name: x, dtype: timedelta64[ns]
```

### PR DESCRIPTION
**Pull request by jbrockmendel**
Tests will be added after #19139 is merged, since largely that will involve removing an "xfail" from tests that already exist there.

- [x] closes #19158
- [ ] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.23.0.txt [hunk 1]
```diff
@@ -386,9 +386,7 @@ Conversion
 - Bug in :class:`WeekOfMonth` and :class:`LastWeekOfMonth` where default keyword arguments for constructor raised ``ValueError`` (:issue:`19142`)
 - Bug in localization of a naive, datetime string in a ``Series`` constructor with a ``datetime64[ns, tz]`` dtype (:issue:`174151`)
 - :func:`Timestamp.replace` will now handle Daylight Savings transitions gracefully (:issue:`18319`)
-
-
```
Comment (jreback) [inline doc/source/whatsnew/v0.23.0.txt]: pls leave white space this is on purpose to avoid conflicts

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.23.0.txt (hunk 1 above)
Comment (jbrockmendel) [reply doc/source/whatsnew/v0.23.0.txt]: Gotcha, will fix.

**Comment by jreback** [comment]
#19139 merged, so rebase and remove xfails.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. ping on green.

**Comment by jbrockmendel** [comment]
Ping

**Comment by jreback** [comment]
thanks!

**Pull request merged**